docs/examples/example_onnx.ipynb

```diff
@@ -0,0 +1,337 @@
+   "metadata": {},
+   "source": [
+    "### Converting the base models to ONNX\n",
+    "\n",
+    "Before being able to convert the MetaLearner to ONXX we need to manually convert the necessary\n",
+    "base models for the prediction. To get a list of the necessary base models that need to be\n",
+    "converted we can use :meth:`~metalearners.MetaLearner.necessary_onnx_models`."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "xlearner.necessary_onnx_models()"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "We see that we need to convert the ``\"propensity_model\"``, the ``\"control_effect_model\"``\n",
+    "and the ``\"treatment_effect_model\"``. We can do this with the following code where we\n",
+    "use the ``convert_lightgbm`` function from the ``onnxmltools`` package.\n",
+    "\n",
+    "```{note}\n",
+    "It is important to know that for classifiers we need to pass the ``zipmap=False`` option. This\n",
+    "is required so the output probabilities are a Matrix and not a list of dictionaries.\n",
+    "In the case of using a ``sklearn`` model and using the ``convert_sklearn`` function, this\n",
+    "option needs to be specified with the ``options={\"zipmap\": False}`` parameter.\n",
+    "```"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import onnx\n",
+    "from onnxmltools import convert_lightgbm\n",
+    "from onnxconverter_common.data_types import FloatTensorType\n",
+    "\n",
+    "onnx_models: dict[str, list[onnx.ModelProto]] = {\n",
+    "    \"control_effect_model\": [],\n",
+    "    \"treatment_effect_model\": [],\n",
+    "    \"propensity_model\": [],\n",
+    "}\n",
+    "\n",
+    "for model in xlearner._nuisance_models[\"propensity_model\"]:\n",
+    "    onnx_model = convert_lightgbm(\n",
+    "        model._overall_estimator,\n",
+    "        initial_types=[(\"X\", FloatTensorType([None, len(feature_columns)]))],\n",
+    "        zipmap=False,\n",
+    "    )\n",
+    "    onnx_models[\"propensity_model\"].append(onnx_model)\n",
+    "\n",
+    "for model in xlearner._treatment_models[\"control_effect_model\"]:\n",
+    "    onnx_model = convert_lightgbm(\n",
+    "        model._overall_estimator,\n",
+    "        initial_types=[(\"X\", FloatTensorType([None, len(feature_columns)]))],\n",
+    "    )\n",
+    "    onnx_models[\"control_effect_model\"].append(onnx_model)\n",
+    "\n",
+    "for model in xlearner._treatment_models[\"treatment_effect_model\"]:\n",
+    "    onnx_model = convert_lightgbm(\n",
+    "        model._overall_estimator,\n",
+    "        initial_types=[(\"X\", FloatTensorType([None, len(feature_columns)]))],\n",
+    "    )\n",
+    "    onnx_models[\"treatment_effect_model\"].append(onnx_model)\n",
+    "\n",
+    "onnx_model = xlearner.build_onnx(onnx_models)"
```

Comment:
Might be worth commenting a bit on what `build_onnx` does. As I see it, it combines the models into a single artifact. The combined model has a single, 2D input named `"X"` and a single output named `"tau"`.